RAD-5637-add-event-detectors-queries-for-h2db

juancarlosgl opened the pull request:
RAD-5637/add-event-detectors-queries-for-h2db

juancarlosgl committed:
RAD-5637 add event detectors queries for h2db

aasbra commented:
@juancarlosgl Even though existingDurationType is not required for the script to work, we added it to the sample MySQL EDIT script. We should probably add the existingDurationType column to the MariaDB EDIT sample too.

aasbra commented on event_detector_scripts/create-event-detectors.js:
```diff
@@ -543,10 +557,14 @@ console.log(message);
+    )
+ AND
+    (
+        dp.JSON_UNQUOTE(JSON_EXTRACT(eD.data, '$.alarmLevel')) IN ('NONE', 'INFORMATION', 'IMPORTANT', 'WARNING', 'URGENT', 'CRITICAL', 'LIFE_SAFETY', 'DO_NOT_LOG', 'IGNORE')
```
For MySQL JSON, we use the ->> notation in all the other scripts.
This: 
_eD.data->>'$.alarmLevel'_
is simpler and easier to read than:
_dp.JSON_UNQUOTE(JSON_EXTRACT(eD.data, '$.alarmLevel')) IN ('NONE', 'INFORMATION', 'IMPORTANT', 'WARNING', 'URGENT', 'CRITICAL', 'LIFE_SAFETY', 'DO_NOT_LOG', 'IGNORE')_

MariaDB doesn't support this, so we should keep 2 sample scripts, instead of trying to make 1 universal script. Most users are running MySQL, so we don't need to make those scripts harder to read to add support for MariaDB.

juancarlosgl replied:
> For MySQL JSON, we use the ->> notation in all the other scripts. This: _eD.data->>'$.alarmLevel'_ is simpler and easier to read than: _dp.JSON_UNQUOTE(JSON_EXTRACT(eD.data, '$.alarmLevel')) IN ('NONE', 'INFORMATION', 'IMPORTANT', 'WARNING', 'URGENT', 'CRITICAL', 'LIFE_SAFETY', 'DO_NOT_LOG', 'IGNORE')_
> 
> MariaDB doesn't support this, so we should keep 2 sample scripts, instead of trying to make 1 universal script. Most users are running MySQL, so we don't need to make those scripts harder to read to add support for MariaDB.

All right, leaving it as eD.data->>'$.alarmLevel'

aasbra commented on event_detector_scripts/create-event-detectors.js:
```diff
@@ -514,7 +528,7 @@ console.log(message);
 /*
 
- -- The following SQL query is suggested to create the CSV file for Mysql or MariaDB databases
+ -- The following SQL query is suggested to create the CSV file for Mysql and MariaDB
```
```suggestion
 -- The following SQL query is suggested to create the CSV file for Mysql
```

juancarlosgl replied:
Done

aasbra commented on event_detector_scripts/create-event-detectors.js:
```diff
@@ -543,10 +557,14 @@ console.log(message);
 
 
- -- Another example also for both databases(Mysql/MariaDB)
+ -- Another example also for Mysql and MariaDB
```
```suggestion
 -- This example is for MariaDB
```

juancarlosgl replied:
Done

aasbra committed:
Update edit-event-detectors.js

Fixed bug where ${type} is not defined.

aasbra committed:
Update delete-event-detectors.js

Fixed reference to RESERVED_EMPTY and updated validations to match current standards in other scripts.

juancarlosgl committed:
RAD-5637 addressed PR commends

juancarlosgl commented:
> @juancarlosgl Even though existingDurationType is not required for the script to work, we added it to the sample MySQL EDIT script. We should probably add the existingDurationType column to the MariaDB EDIT sample too.

Right, I added it on edit-event-detectors.js

aasbra commented on event_detector_scripts/delete-event-detectors.js:
```diff
@@ -74,34 +75,38 @@ let count = 0;
     }
 
     try {
         compare(detector.getId(), eventDetectorCsv.eventDetectorId, "eventDetectorId MISMATCH");
```
```suggestion
        compare(detector.getId(), Number(eventDetectorCsv.eventDetectorId), "eventDetectorId MISMATCH");
```
Convert the value to a Number for the compare function.

juancarlosgl committed:
RAD-5637 Update event_detector_scripts/delete-event-detectors.js

Co-authored-by: Aaron Asbra <31995798+aasbra@users.noreply.github.com>
